### DISCUSSION TITLE
WASM Import in @vercel/og

### DISCUSSION THREAD

**Discussion by swamidass** [remix-run/remix#4439] [Q&A]
When I try to create a route using @vercel/og in remix, I get this error:

[1]  ✘ [ERROR] No loader is configured for ".wasm" files: node_modules/@vercel/og/vendor/yoga.wasm?module
[1] 
[1]     node_modules/@vercel/og/dist/index.js:1:164:
[1]       1 │ ...ule";import T from"../vendor/yoga.wasm?module";var v=String.from...
[1]         ╵                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

It would be great to use their open-graph image generation code in remix. What is the current work around to solve this compile error?

**Comment by machour** [comment]
Here's an example of how a loader was added for the zip file extension: #3985
Would you like to open a pull request against the `dev` branch adding support for wasm?

**Reply by MichaelDeBoey** [reply]
We already have #3299 to support `.wasm` files

**Comment by benjaminleonard** [comment]
I'm using the nightly with `.wasm` support, and added the following to my `remix.config.js` file but I still get the following error:

```
✘ [ERROR] No loader is configured for ".wasm" files: node_modules/@vercel/og/vendor/resvg.simd.wasm?module

    node_modules/@vercel/og/dist/index.js:1:116:
      1 │ ...";import N from"../vendor/resvg.simd.wasm?module";import T from"...
        ╵                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

 ✘ [ERROR] No loader is configured for ".wasm" files: node_modules/@vercel/og/vendor/yoga.wasm?module

    node_modules/@vercel/og/dist/index.js:1:164:
      1 │ ...ule";import T from"../vendor/yoga.wasm?module";var v=String.from...
        ╵     
```

**Reply by machour** [reply]
What nightly are you using ?

**Reply by benjaminleonard** [reply]
```
"@remix-run/dev": "^0.0.0-nightly-929d847-20221111",
"@remix-run/serve": "^0.0.0-nightly-929d847-20221111",
```

Installed this based on the comment in the PR

**Reply by machour** [reply]
Could your try 1.8.0 ? Published a few hours ago

**Reply by benjaminleonard** [reply]
Thanks, I get a slightly different error:

```
/Users/leonard/Desktop/Development/oxide-computer/node_modules/yoga-wasm-web/dist/index.js:1
import i from"./entry.js";import o from"./yoga.mjs";function s(e,t){return t}async function c(e){let t=await o({instantiateWasm(r,a){return WebAssembly.instantiate(e,r).then(n=>{a(n.instance||n)}),{}},locateFile(){return""}});return i(s,t)}async function l(e){let t=await o({instantiateWasm(r,a){return WebAssembly.instantiateStreaming(e,r).then(n=>{a(n.instance||n)}),{}},locateFile(){return""}});return i(s,t)}export{c as default,l as initStreaming};
                                                                                                                                                        ^
TypeError: WebAssembly.instantiate(): Argument 0 must be a buffer source or a WebAssembly.Module object
    at Object.instantiateWasm (/Users/leonard/Desktop/Development/oxide-computer/node_modules/yoga-wasm-web/dist/index.js:1:153)
    at /Users/leonard/Desktop/Development/oxide-computer/node_modules/yoga-wasm-web/dist/yoga.mjs:65:421
    at /Users/leonard/Desktop/Development/oxide-computer/node_modules/yoga-wasm-web/dist/yoga.mjs:66:356
    at c (/Users/leonard/Desktop/Development/oxide-computer/node_modules/yoga-wasm-web/dist/index.js:1:110)
    at Object.<anonymous> (/Users/leonard/Desktop/Development/oxide-computer/node_modules/@vercel/og/src/og.ts:17:25)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
ERROR: "dev:remix" exited with 1.
```

**Reply by machour** [reply]
@benwis any idea what might be causing this? 🙏🏼 
@benjaminleonard would be great if you provided a public repo reproducing the error. 
If something is still wrong with wasm support let's open a new issue.

**Reply by benjaminleonard** [reply]
Done, issue here: https://github.com/remix-run/remix/issues/4735 and repo here: https://github.com/benjaminleonard/vercel-og-wasm

**Comment by machour** [comment]
@swamidass can you give 1.8.0 a try ?

**Reply by swamidass** [reply]
I will try and look, but it will be a few weeks. For reference, this is the library that was failing to work:

https://github.com/rdkit/rdkit-js

The code there is largely (exclusively?) client side, and works fine. Convert any of those functions into server-side, and it won't work. That's the problem. And a frustrating one too.

In the end, I built the website (https://xenosite.org/) but had to put the rdkit logic on another (persistent) server, unfortunately.  

Did you figure out what was the problem?

**Comment by swamidass** [comment]
Another use case (probably more broadly useful) is the Vercel WASM rendering to create og images on the fly in an edge function:

https://vercel.com/templates/next.js/og-image-generation
https://github.com/vercel/og-image

I'd really like to add this to my remix project, but it failed miserably. :( Has anyone had any luck getting it to work?

I really love some aspects of ReMIX but these two issues together, along with the big changes in NextJs 13, have me on the fence.
